flow/Histogram.cpp

```diff
@@ -45,7 +45,7 @@ static HistogramRegistry* globalHistograms = nullptr;
 #pragma region HistogramRegistry
 
 HistogramRegistry& GetHistogramRegistry() {
-	ISimulator::ProcessInfo* info = g_simulator.getCurrentProcess();
+	ISimulator::ProcessInfo* info = g_network && g_network->isSimulated() ? g_simulator.getCurrentProcess() : nullptr;
```

Comment:
I think that's a question for @sears

Reply:
What did UBSAN find here?  I just noticed that g_simulator is likely nullptr, which only works because getCurrentProcess only accesses static variables.  That seems wrong.

While we're looking at this, any idea why:
`static thread_local ProcessInfo* currentProcess;`
in simulator.h is static?  It seems fishy to me:
```
	Sim2()
	  : time(0.0), timerTime(0.0), taskCount(0), yielded(false), yield_limit(0), currentTaskID(TaskPriority::Zero) {
		// Not letting currentProcess be nullptr eliminates some annoying special cases
		currentProcess =
		    new ProcessInfo("NoMachine",
		                    LocalityData(Optional<Standalone<StringRef>>(), StringRef(), StringRef(), StringRef()),
		                    ProcessClass(),
		                    { NetworkAddress() },
		                    this,
		                    "",
		                    "");
```
so, it's static, but we overwrite it each time the simulator is instantiated.  I guess it means that bugs like the one I wrote will work, but is there any other reason for it to be static?

I checked all the invocations of getHistogram and GetHistogramRegistry.  At first glance, they all are happening after the simulator should be up and running.

If this were invoked in simulation before the network was created then some histograms would not be logged, which would be difficult to debug.  The existing code was vulnerable to the same race, so don't feel obligated to improve the behavior if that happens.

Reply:
UBSAN found "applying non-zero offset to null pointer". So it's not allowed to even call getCurrentProcess here outside of simulation, which is the main problem this intends to address.